extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -61,34 +66,101 @@ namespace nifi {
 namespace minifi {
 namespace processors {
 
-core::Property TailFile::FileName("File to Tail", "Fully-qualified filename of the file that should be tailed", "");
+core::Property TailFile::FileName("File to Tail", "Fully-qualified filename of the file that should be tailed when using single file mode, or a file regex when using multifile mode", "");
 core::Property TailFile::StateFile("State File", "Specifies the file that should be used for storing state about"
                                    " what data has been ingested so that upon restart NiFi can resume from where it left off",
                                    "TailFileState");
 core::Property TailFile::Delimiter("Input Delimiter", "Specifies the character that should be used for delimiting the data being tailed"
                                    "from the incoming file.",
                                    "");
+
+core::Property TailFile::TailMode(
+    core::PropertyBuilder::createProperty("tail-mode", "Tailing Mode")->withDescription("Specifies the tail file mode. In 'Single file' mode only a single file will be watched. "
+                "In 'Multiple file' mode a regex may be used. Note that in multiple file mode we will still continue to watch for rollover on the initial set of watched files. "
+                "The Regex used to locate multiple files will be run during the schedule phrase. Note that if rotated files are matched by the regex, those files will be tailed.")->isRequired(true)->withAllowableValue<std::string>("Single file")->withAllowableValue("Multiple file")->withDefaultValue(
```

Comment:
Just nitpicking: can you add the properties after the description to a new line? Just to make them appear on the screen by default. :)